<a href="https://colab.research.google.com/github/FarahRamadhaniPutri/CaseStudy_Data_Wrangling_Kelompok_B/blob/main/Casestudy_data_wrangling_Kelompok_B.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Group B
1. Timothy Devan Chandra (1906375726)
2. Nadia Sukesi Sianipar (2106700776)
3. Abdul Muhaimin (2106703052)
4. Aulia Shinta Madani (2106724984)
5. Farah Ramadhani Putri (2106654832)
6. Daffa Al Ghifary (2006463420)

<center><h1><strong><font color="blue">Studi Kasus Data Wrangling</font></strong></h1></center>

![taudata Analytics](https://github.com/taudataid/eLearning/raw/master/images/covers/cover_ui-n-taudata.jpg "taudata Analytics")

In [ ]:
import warnings; warnings.simplefilter('ignore')
import pandas as pd

try: # Running Locally, yakinkan "file_" berada di folder "data"
    mengajar = pd.read_csv('data/mengajar.csv', encoding='utf8')
    dosen = pd.read_csv('data/tb_dosen.csv', encoding='utf8')
    kelas = pd.read_csv('data/tb_kelas.csv', encoding='utf8')
    krs = pd.read_csv('data/tb_krs.csv', encoding='utf8')
    mahasiswa = pd.read_csv('data/tb_mahasiswa.csv', encoding='utf8')
    matkul = pd.read_csv('data/tb_mata_kuliah.csv', encoding='utf8')
    nilai = pd.read_csv('data/tb_nilai.csv', encoding='utf8')
except: # Running in Google Colab
    !mkdir data
    !wget -P data/ https://raw.githubusercontent.com/taudataid/eLearning/master/data/tb_nilai.csv
    !wget -P data/ https://raw.githubusercontent.com/taudataid/eLearning/master/data/mengajar.csv
    !wget -P data/ https://raw.githubusercontent.com/taudataid/eLearning/master/data/tb_dosen.csv
    !wget -P data/ https://raw.githubusercontent.com/taudataid/eLearning/master/data/tb_kelas.csv
    !wget -P data/ https://raw.githubusercontent.com/taudataid/eLearning/master/data/tb_krs.csv
    !wget -P data/ https://raw.githubusercontent.com/taudataid/eLearning/master/data/tb_mahasiswa.csv
    !wget -P data/ https://raw.githubusercontent.com/taudataid/eLearning/master/data/tb_mata_kuliah.csv
    mengajar = pd.read_csv('data/mengajar.csv', encoding='utf8')
    dosen = pd.read_csv('data/tb_dosen.csv', encoding='utf8')
    kelas = pd.read_csv('data/tb_kelas.csv', encoding='utf8')
    krs = pd.read_csv('data/tb_krs.csv', encoding='utf8')
    mahasiswa = pd.read_csv('data/tb_mahasiswa.csv', encoding='utf8')
    matkul = pd.read_csv('data/tb_mata_kuliah.csv', encoding='utf8')
    nilai = pd.read_csv('data/tb_nilai.csv', encoding='utf8')

mengajar.shape, dosen.shape, kelas.shape, krs.shape, mahasiswa.shape, matkul.shape, nilai.shape

((14, 4), (4, 6), (4, 4), (7, 6), (15, 8), (9, 3), (33, 7))

In [ ]:
# Code untuk Soal No 01 diberikan sebagai Contoh dan Bonus
# Pelajari dengan baik

nilai_dosen = nilai.groupby(nilai['NIDN']).mean()[['nilai']]
nilai_dosen = nilai_dosen.rename({'nilai': 'mean_nilai'}, axis=1) 
nilai_dosen = pd.merge(dosen, nilai_dosen, left_on='NIDN', right_index=True)
nilai_dosen = nilai_dosen[['NIDN', 'nama', 'mean_nilai']]
sb = nilai.groupby(nilai['NIDN']).std()[['nilai']]
sb = sb.rename({'nilai': 'std_nilai'}, axis=1) 
nilai_dosen = nilai_dosen.set_index(['NIDN']) 
nilai_dosen = pd.concat([nilai_dosen, sb], axis=1)
nilai_dosen

nama  mean_nilai  std_nilai
NIDN                                      
123    Andri Basuki   85.750000   7.741355
234        zulfikar   80.428571  15.883504
787            yudi   70.571429  13.010984
345   Annisa Fitria   60.090909  16.220077

# Interpretasi 01: 

* Dapat terlihat bahwa rata-rata nilai dosen paling tinggi yaitu yang diberikan oleh Andri Basuki = 85.750000, sedangkan jika dibandingkan dengan rata-rata yang diberikan Annisa Fitria = 60.090909. Hal ini menunjukkan adanya perbedaan rata-rata antar dosen yang cukup jauh. 
* Dapat terlihat pula bahwa standar deviasi nilai antar dosen rentangnya cukup jauh, sehingga dapat dikatakan nilai yang diberikan oleh dosen cukup bervariasi. 

Maka, dapat disimpulkan terdapat dosen yang cenderung memberikan nilai lebih tinggi yaitu Andri Basuki atau lebih rendah yaitu Annisa Fitria jika dibandingkan dengan dosen lainnya

In [ ]:
# Code Untuk Soal No 02

pendidikan_dosen = pd.merge(dosen, nilai, on='NIDN')
pendidikan_dosen = pendidikan_dosen[['Pendidikan_Terakhir','nilai']]
mean_nilai = pendidikan_dosen.groupby(pendidikan_dosen['Pendidikan_Terakhir']).mean()[['nilai']]
mean_nilai = mean_nilai.rename({'nilai':'mean_nilai'}, axis = 1)
std_nilai = pendidikan_dosen.groupby(pendidikan_dosen['Pendidikan_Terakhir']).std()[['nilai']]
std_nilai = std_nilai.rename({'nilai':'std_nilai'}, axis = 1)
pendidikan_dosen = pd.concat([mean_nilai, std_nilai], axis = 1)
pendidikan_dosen

mean_nilai  std_nilai
Pendidikan_Terakhir                       
S2                    64.166667  15.561075
S3                    83.266667  12.068061

# Interpretasi 02:
* Pertama kita membuat variabel 'pendidikan_dosen' untuk menggabungkan tabel dosen dan nilai berdasarkan 'NIDN' kemudian hanya diambil kolom 'Pendidikan_Terakhir' dan 'Nilai'
* Selanjutnya akan dicari mean (dengan variabel mean_nilai) dan standar deviasidengan variabel std_nilai dari nilai pada tabel pendidikan_dosen dengan menggunakan fungsi grouby
* Kemudian rename kolom 'Nilai' dengan dengan 'mean_nilai' dan 'std_nilai'
* Gabungkan tabel mean_nilai dan std_nilai
* Dapat terlihat bahwa rata-rata nilai yang diberikan dosen dengan pendidikan terakhir S3 lebih tinggi daripada dosen dengan pendidikan terkahir S2 dengan rentang rata-rata nya sekitar 19, hal ini cukup jauh.
* Kesimpulan yang didapat adalah terdapat perbedaan dalam pemberian nilai antara dosen dengan pendidikan terakhir S3 dan S2

In [ ]:
# Code Soal No 03

nilai_mahasiswa = nilai.groupby(nilai['nim']).mean()[['nilai']]
nilai_mahasiswa = nilai_mahasiswa.rename({'nilai':'mean_nilai'}, axis=1)
nilai_mahasiswa = pd.merge(mahasiswa, nilai_mahasiswa, left_on='nim', right_index=True)
nilai_mahasiswa = nilai_mahasiswa[['nim', 'Asal_Sekolah', 'mean_nilai']]
sb_m = nilai.groupby(nilai['nim']).std()[['nilai']]
sb_m = sb_m.rename({'nilai':'std_nilai'}, axis=1)
nilai_mahasiswa = nilai_mahasiswa.set_index(['nim'])
nilai_mahasiswa = pd.concat([nilai_mahasiswa, sb_m], axis=1)
nilai_mahasiswa.pivot_table(index='Asal_Sekolah')

mean_nilai  std_nilai
Asal_Sekolah                       
MA             71.937500  19.673045
SMK            85.900000   4.693409
SMU            66.416667  16.729164

# Interpretasi 03:
* Dapat dilihat bahwa rata-rata nilai mahasiswa paling tinggi pada mahasiswa yang berasal dari SMK yaitu sebesar 85.900000. Sedangkan rata-rata nilai mahasiswa paling rendah pada mahasiswa yang berasal dari SMU yaitu sebesar 66.416667. Dapat disimpulkan bahwa terdapat perbedaan performa akademik antara mahasiswa yang berasal dari Madrasah/SMK dan SMU. Perfroma akademik mahasiswa yang berasal dari Madrasah/SMK relatif lebih baik dibanding SMU.

* Dapat diketahui bahwa std nilai rentangnya cukup jauh, sehingga dapat dikatakan nilai yang terdapat pada mahasiswa  yang berasal dari MA/SMK dan SMU rentangnya cukup bervariasi.

Saran: Apabila kamu seorang murid SMP yang ingin berkuliah. Lanjutkan jenjang pendidikan kamu ke SMK. Agar memiliki peluang nilai yang lebih tinggi saat menjadi mahasiswa.

In [ ]:
# Code Soal No 04

nilai_mahasiswa = nilai.groupby(nilai['nim']).mean()[['nilai']]
nilai_mahasiswa = nilai_mahasiswa.rename({'nilai':'mean_nilai'}, axis=1)
nilai_mahasiswa = pd.merge(mahasiswa, nilai_mahasiswa, left_on='nim', right_index=True)
nilai_mahasiswa = nilai_mahasiswa[['nim', 'Gender', 'mean_nilai']]
sb_m = nilai.groupby(nilai['nim']).std()[['nilai']]
sb_m = sb_m.rename({'nilai':'std_nilai'}, axis=1)
nilai_mahasiswa = nilai_mahasiswa.set_index(['nim'])
nilai_mahasiswa = pd.concat([nilai_mahasiswa, sb_m], axis=1)
nilai_mahasiswa.pivot_table(index='Gender')

mean_nilai  std_nilai
Gender                       
Pria     75.750000  15.905278
Wanita   72.333333  11.001154

# Interpretasi 04

* Berdasarkan data yang diperoleh. Dapat disimpulkan bahwa tidak ada perbedaan yang signifikan antara kemampuan akademik mahasiswa dan mahasiswi. Hal ini karena rata rata nilai mahasiswa yaitu sebesar 75.75 tidak jauh berbeda dengan rata rata nilai mahasiswi yaitu 72.33.

* Dapat diketahui bahwa std nilai rentangnya cukup jauh, sehingga dapat dikatakan nilai yang terdapat pada mahasiswa dan mahasiswi rentangnya cukup bervariasi.

In [ ]:
# Code soal No 05

nilai_mahasiswa = nilai.groupby(nilai['nim']).mean()[['nilai']]
nilai_mahasiswa = nilai_mahasiswa.rename({'nilai': 'mean_nilai'}, axis=1)
nilai_mahasiswa = pd.merge(mahasiswa, nilai_mahasiswa, left_on='nim', right_index=True)
nilai_mahasiswa = nilai_mahasiswa[['nim', 'Angkatan', 'mean_nilai']]
sb = nilai.groupby(nilai['nim']).std()[['nilai']]
sb = sb.rename({'nilai': 'std_nilai'}, axis=1)
nilai_mahasiswa = nilai_mahasiswa.set_index(['nim'])
nilai_mahasiswa = pd.concat([nilai_mahasiswa, sb], axis=1)
nilai_Angkatan = nilai_mahasiswa.groupby(nilai_mahasiswa['Angkatan']).mean()[['mean_nilai']]
sb = nilai_mahasiswa.groupby(nilai_mahasiswa['Angkatan']).std()[['mean_nilai']]
sb = sb.rename({'mean_nilai': 'std_nilai'}, axis=1)
sb
nilai_AngkatanStd = pd.merge(nilai_Angkatan, sb, left_on='Angkatan', right_index=True)
nilai_AngkatanStd

mean_nilai  std_nilai
Angkatan                       
2019        75.59375  13.433658
2020        73.00000  15.044379

# Interpretasi 05

* Pertama-tama kita membuat variabel 'nilai_mahasiswa' untuk menggabungkan antara nim dan nilai kemudian dihitung mean nilai dan std nilai setiap mahasiswa.
* Lalu kita gunakan fungsi groupby angkatan untuk mengelompokkan per angkatan.
* Terlihat bahwa rata-rata mean nilai antara angkatan 2019 dan 2020 tidak berbeda jauh. Begitu pula dengan std nilainya.
* Dapat diketahui bahwa std nilai rentangnya cukup jauh, sehingga dapat dikatakan nilai yang terdapat pada mahasiswa di setiap angkatannya rentangnya cukup bervariasi.
* Sehingga dapat disimpulkan bahwa tidak terdapat perbedaan prestasi akademik yang nampak di data antar angkatan mahasiswa.

In [ ]:
# Code soal No 06

nilai_mahasiswa = nilai.groupby(nilai['nim']).mean()[['nilai']]
nilai_mahasiswa = nilai_mahasiswa.rename({'nilai': 'mean_nilai'}, axis=1) 
nilai_mahasiswa = pd.merge(mahasiswa, nilai_mahasiswa, left_on='nim', right_index=True)
nilai_mahasiswa = nilai_mahasiswa[['nim', 'Asal_Daerah', 'mean_nilai']]
sb_1 = nilai.groupby(nilai['nim']).std()[['nilai']]
sb_1 = sb_1.rename({'nilai': 'std_nilai'}, axis=1) 
nilai_mahasiswa = nilai_mahasiswa.set_index(['nim']) 
nilai_mahasiswa = pd.concat([nilai_mahasiswa, sb_1], axis=1)
nilai_mahasiswa.pivot_table(index='Asal_Daerah')

mean_nilai  std_nilai
Asal_Daerah                       
Lokal         78.000000  10.283147
Luar Daerah   68.958333  17.891313

# Interpretasi 06

* Dapat terlihat bahwa rata-rata nilai dari mahasiswa yang berasal dari lokal dan luar daerah memiliki perbedaan yang jauh, yaitu rata- rata mahasiswa lokal = 78, sedangkan rata-rata mahasiswa luar daerah = 68.958333. 
* Kemudian terlihat juga bahwa standar deviasi baik dari mahasiswa lokal maupun luar daerah memiliki standar deviasi yang cukup besar sehingga nilai dari setiap mahasiswanya dapat dikatakan cukup bervariasi.

Oleh karena itu dapat disimpulkan terdapat perbedaan performa akademik antara mahasiswa berdasarkan asal daerahnya yaitu mahasiswa lokal dengan mahasiswa luar daerah.

#Kesimpulan

1. Apakah ada dosen yang cenderung memberikan nilai cenderung lebih tinggi (atau lebih rendah) dibandingkan dosen lainnya?

Ya, dapat dilihat perbandingannya yaitu Dosen Andri Basuki memiliki rata-rata nilai 85.75. Namun, nilai yang diberikan Dosen Andri Basuki kurang bervariasi karena standar deviasi yang paling rendah jika dibandingkan dengan dosen lainnya. Sedangkan Dosen Annisa Fitria memiliki rata-rata paling rendah yaitu 60. Tetapi nilai yang diberikan bervariasi dibandingkan dosen lainnya, karena standar deviasi Dosen Annisa Fitria paling tinggi diantara dosen lainnya yaitu 16.22.

2. Menurut data yang ada, apakah terdapat perbedaan dalam pemberian nilai antara dosen dengan Pendidikan Terakhir S3 dan S2?

Ya, dapat dilihat perbandingan nilai dosen dengan pendidikan terakhir S3 yaitu 83.266667 dibandingkan dengan pendidikan terakhir s2 yaitu 64.166667. Terlihat bahwa nilai dosen dengan pendidikan terakhir S3 lebih tinggi daripada S2. Namun, jika dilihat dari standar deviasinya, nilai yang diberikan oleh dosen dengan pendidikan terakhir S2 lebih bervariasi daripada dosen S3.

3. Apakah terdapat perbedaan performa akademik antara mahasiswa yang berasal dari Madrasah/SMK dan SMU?

Ya, Perfroma akademik mahasiswa yang berasal dari Madrasah/SMK relatif lebih baik dibanding SMU. Terlihat pada rata-rata nilai mahasiswa paling tinggi pada mahasiswa yang berasal dari SMK yaitu sebesar 85.900000. Sedangkan rata-rata nilai mahasiswa paling rendah pada mahasiswa yang berasal dari SMU yaitu sebesar 66.416667.

4. Apakah ada perbedaan yang nampak di data terhadap kemampuan akademik antara mahasiswa dan mahasiswi?

Tidak ada perbedaan yang signifikan. Namun, tidak dapat dipungkiri, terdapat perbedaan yg cukup kecil antara nilai mahasiswa dan mahasiwi. Beda nya sekitar +- 3 poin pada rata-ratanya.

5. Apakah ada perbedaan prestasi akademik yang nampak di data antar angkatan mahasiswa?

Tidak ada perbedaan yang signifikan. Namun, tidak dapat dipungkiri, terdapat perbedaan yg cukup kecil antara nilai mahasiswa dan mahasiwi. Beda nya sekitar +- 2.5 poin pada rata-ratanya.

6. Apakah ada permasalahan di data atau informasi lain yang anda temukan di data?

Terdapat informasi baru yaitu dapat terlihat bahwa rata-rata nilai dari mahasiswa yang berasal dari lokal dan luar daerah memiliki perbedaan yang jauh, yaitu rata- rata mahasiswa lokal = 78, sedangkan rata-rata mahasiswa luar daerah = 68.958333. Maka, terdapat perbedaan performa akademik antara mahasiswa berdasarkan asal daerahnya yaitu mahasiswa lokal dengan mahasiswa luar daerah.